protocol/eth-shared-lockbox.md

```diff
@@ -35,125 +35,75 @@ OP Chains will be governed within a common Chain Cluster governance entity (the
-```
-
-Adding a new chain can be done by introducing a new function called `addChain`. The process would look as follows:
-
-1. The `updater` calls the function with `chainId` and `SystemConfig`.
-2. An external call is made to each `SystemConfig` from already added chains with the new `chainId`.
-3. The new `SystemConfig` is called to add all previously existing `chainId` from `SuperchainConfig` to itself.
-4. Each call concludes by emitting a `DepositedTransaction` event.
-5. The `OptimismPortal` address is stored in the `Sharedlockbox` mapping.
-6. Each chain asynchronously includes each new deposit and updates `dependencySet` in `L1Block`.
-
-The [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry) or the [OP Contracts Manager](https://specs.optimism.io/experimental/op-contracts-manager.html?highlight=chain#chain-id-source-of-truth) can be used as the source of truth to add chains that have been verified beforehand as compatible.
-
-A code example would look like this:
-
-```solidity
-function addChain(uint256 _chainId, address _systemConfig) external {
-  require(msg.sender == updater(), "Unauthorized");
-  // Add to the dependency set and check it is not already added (`add()` returns false if it already exists)
-  require(_dependencySet.add(_chainId), "Chain already added");
-
-  // Store the system config
-  systemConfigs[_chainId] = _systemConfig;
+### `SharedLockbox` implementation
 
-  // Loop through the dependency set and update the dependency for each chain
-  for (uint256 i; i < _dependencySet.length() - 1; i++) {
+A minimal set of functions should include:
 
-    uint256 currentId = _dependencySet.at(i);
+- `lockETH`: Accepts ETH from the `depositTransaction` originating from a valid `OptimismPortal`.
+- `unlockETH`: Releases ETH from the `finalizeWithdrawalTransaction` originating from a valid `OptimismPortal`.
 
-    // Skip recently added chain
-    if (_chainId == currentId) continue;
+Access control for `lockETH` and `unlockETH` is validated against the mapping of authorized `OptimismPortal` addresses.
 
-    systemConfigs[currentId].addDependency(_chainId);
+The upgrade controller role can add `OptimismPortal` addresses to the `SharedLockbox` contract.
 
-    systemConfigs[_chainId].addDependency(currentId);
-  }
+### `OptimismPortal` upgrade process
 
-  address portal = _systemConfig.optimismPortal();
+A permissionless ETH migration function can be added to the `OptimismPortal` to allow transferring all ETH to the `SharedLockbox` at any time. The `SharedLockbox` address is set during initialization of the `OptimismPortal` and cannot be changed afterwards.
 
-  // Authorize the portal on the shared lockbox
-  SHARED_LOCKBOX.authorizePortal(portal);
+The migration process would:
 
-  emit ChainAdded(_chainId, _systemConfig, portal);
-}
-```
+- Allow any caller to trigger the migration
+- Transfer the entire ETH balance of the `OptimismPortal` to the pre-configured `SharedLockbox`
```

Comment:
Rather than add a new function, it might be simpler to just do this during the upgrade process, via an `upgrade()` function.

Reply:
That's an implementation detail tho

Reply:
Let's use a function